src/components/Button/Button.tsx

```diff
@@ -1,42 +1,46 @@
-import { ClickHandler, useLoading } from 'hooks/useLoading';
-import React, { useRef } from 'react';
+import React from 'react';
 import { CommonButtonProps } from 'utils/common';
 import { TestProps } from 'utils/types';
 
+import buttonTokens from '../Button/Button.tokens';
 import ButtonBase, { ButtonBaseProps } from '../ButtonBase/ButtonBase';
-import { buttonSpanStyle, childrenWrapperStyle, iconStyle } from './Button.style';
-import ButtonLoader from './ButtonLoader';
+import { buttonSpanStyle } from './Button.style';
+import Avatar, { AvatarProps } from 'components/Avatar';
+import Icon, { AcceptedIconNames } from 'components/Icon';
 
-export type onClickProp = { onClick: ClickHandler };
-export type ButtonProps = Omit<ButtonBaseProps, 'onClick'> &
+export type ButtonProps = ButtonBaseProps &
   TestProps &
-  onClickProp &
-  CommonButtonProps;
+  CommonButtonProps & {
+    /** An optional icon to put on the right of the button */
```

Comment:
why we moved this from `ButtonBaseProps` to here

Reply:
it seemed to me better to move the leftIcon, rightIcon and avatar props to the Button component since:
1. they are no longer rendered inside the ButtonBase (so there is no point picking them from the ButtonBase for the types declaration)
2. semantically speaking: these props are only included in the Button variation of the ButtonBase; ButtonBase should be more abstract and act more as a wrapper for Button, IconButton or any future Button-like component. Since IconButton declares and renders its icon inside the IconButton, I thought it would be more clear to declare Button's leftIcon, rightIcon and avatar props (and components) inside Button too, alongside its other properties (e.g.: children). 

thoughts on this? 🤔